chainstate/src/detail/mod.rs

```diff
@@ -744,28 +748,38 @@ impl<S: BlockchainStorage, V: TransactionVerificationStrategy> Chainstate<S, V>
         &self.events_controller
     }
 
-    pub fn is_initial_block_download(&self) -> Result<bool, PropertyQueryError> {
-        if self.is_initial_block_download_finished {
-            return Ok(false);
+    pub fn is_initial_block_download(&self) -> bool {
+        !self.is_initial_block_download_finished.test()
+    }
+
+    /// Returns true if the given block timestamp is newer than `ChainstateConfig::max_tip_age`.
+    fn is_fresh_block(&self, time: &BlockTimestamp) -> bool {
+        let now = self.time_getter.get_time();
+        time.as_duration_since_epoch() + self.chainstate_config.max_tip_age.clone().into() > now
+    }
+
+    /// Update `is_initial_block_download_finished` when tip has changes (can only be set once)
+    fn update_initial_block_download_flag(&mut self) -> Result<(), PropertyQueryError> {
+        if self.is_initial_block_download_finished.test() {
+            return Ok(());
         }
 
         // TODO: Add a check for importing and reindex.
 
         // TODO: Add a check for the chain trust.
```

Comment:
I wonder whether these TODOs are still relevant (and if so, what they actually mean).

Reply:
The first one: It's about reindexing. Reindexing is the process of reparsing blocks such that the database can be rebuilt. This is useful in cases of database corruption or when settings like "tx index" are changed. If the reindex process is happening, then it's always true that we're in the "initial block download" state.

The second one: Assuming a monolithic network that we work on during the mainnet, the chain-trust always goes up. Given that, on every release, we can say that a chain-trust below a certain point is always an "initial block download", because we know that the monotonous mainnet must be longer than that point.

Yes, both points are still relevant. But they're more like optimizations than necessities.